File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
             unordered_map<int, int> hashMap; // To store value and its index
    
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i]; // Find the required complement
        if (hashMap.find(complement) != hashMap.end()) {
            // If complement exists in the map, return indices
            return {hashMap[complement], i};
        }
        hashMap[nums[i]] = i; // Store current value with its index
    }
    
    return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer>umap=new HashMap<>();
        int compl;
        for(int i=0;i < nums.length;i++){
            compl=target-nums[i];
            if(umap.containsKey(compl)){
                return new int[] {i,umap.get(compl)};
            }
            else{
                umap.put(nums[i],i);
            }
        }
        return new int[0];
    }
}